src/Engine/TranskribusClient.php

```diff
@@ -0,0 +1,277 @@
+                if($content->{'status'} === 'CREATED'){
+                    $this->processId = $content->{'processId'};
+                    $this->hasError = false;
+                }                
+            }
+        } elseif (401 === $statusCode) {
+            $this->refereshAccessToken();
+            if($this->getHasError()){
+                $message = 'Error Code '.$statusCode.' :: Credentials are incorrect!';
+                $this->setErrorMessage($message);
+            }else{
+                $this->initProcess($imageURL, $config);
+            }
+            
+
+        } else {
+            $rawResponse   = $response->getContent(false);
+            $message = 'Error Code '.$statusCode.' :: Error initiating upload process, try again!';
+            $this->setErrorMessage($message);
+        }
+
+        return $this;
+    }
+
+    public function retrieveProcessStatus(){
+        $url = self::PROCESSES_URL . '/'. $this->processId;
+        $response = $this->httpClient
+        ->request(
+            'GET',
+            $url,
+            [
+                'headers' => [
+                    'Content-Type' => 'application/json',
+                    'Authorization' => 'Bearer '.$this->accessToken
+                ],
+            ]
+        );
+        
+        $statusCode = $response->getStatusCode();
+       
+        if (200 === $statusCode) {
+            $content = json_decode($response->getContent());
+            if (is_null($content)) {
+                $message = 'Transkribus API returned null';
+                $this->setErrorMessage($message);
+            } else {
+                
+                if($content->{'status'} === 'FINISHED'){
+                    $textContent = $content->{'content'};
+                    $this->textResult = $textContent->{'text'};
+                    $this->hasError = false;
+                }                
+            }
+        } elseif (401 === $statusCode) {
+            $this->refereshAccessToken();
+            if($this->getHasError()){
+                $message = 'Error Code '.$statusCode.' :: Credentials are incorrect!';
+                $this->setErrorMessage($message);
+            }else{
+                $this->retryProcessStatus();
+            }
+
+        } else {
+            $rawResponse   = $response->getContent(false);
+            $message = 'Error Code '.$statusCode.' :: Error retrieving process status, try again!';
+            $this->setErrorMessage($message);
+        }
+    }
+
+    public function retryProcessStatus(){
+        $this->retrieveProcessStatus();
+    }
+
+    public function refereshAccessToken(){
+        if(!file_exists('transkribus_credentials.json')){
+            $this->setAccessToken();
+            
+        }else{
+            $data = file_get_contents('transkribus_credentials.json');
+            $content = json_decode($data);
+            $refreshToken = $content->{'refresh_token'};
+            $response = $this->httpClient
```

Comment:
The pattern of making a request, looking at its response code and content, and returning errors, is repeated a few times. It's probably worth abstracting into a separate private function e.g. `TranskribusClient::request()`. That way, the TranskribusAuthCommand could also have TranskribusClient injected and use it to make the requests there.